### ISSUE TITLE
ColumnDef disableExport not working for Print format

### ISSUE THREAD

**Issue by zreecespieces** [mui/mui-x#9983]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Link to live example: https://codesandbox.io/s/cocky-booth-mw47xv?file=/Demo.tsx

Steps:
1. Create basic DataGrid example with columns and some data, this is cloned from documentation sandbox
2. Add disableExport: true to any ColumnDef, on this case I put it on the Full Name column
3. Click export -> Print -> Observe that Full Name column is still present, incorrectly
4. Click export -> Download as CSV -> Observe that the Full Name column is not present, as expected

### Current behavior 😯

Columns marked with disableExport: true are still being displayed when exporting using Print option

### Expected behavior 🤔

Columns marked with disableExport: true should not be displayed when exporting using Print option

### Context 🔦

I'm working on some educational content for MUI and found this behavior which appears to be a bug

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

Codesandbox, Chrome browser (Version 115.0.5790.114), MUI X Data Grid (Version 6.11.0)

**Comment by zreecespieces** [comment]
It seems that adding `slotProps={{ toolbar: { printOptions: { allColumns: true } } }}` causes the disableExport property to take effect. That seems like strange behavior that is unintuitive if intentional.

**Comment by DanailH** [comment]
@zreecespieces I checked the implementation. It's inconsistent compared to the CSV and Exel. It's good that there is a workaround that you can use until we resolve the issue internally.

**Comment by zreecespieces** [comment]
Cool, glad this was a real issue that I was able to identify. I also identified a similar behavior regarding selection and export. In that same [codesandbox](https://codesandbox.io/s/cocky-booth-mw47xv?file=/Demo.tsx):
1. Select a couple of rows using the selection checkbox
2. Click export -> Print -> Observe that all rows are exported, not just selected rows, incorrectly
3. Click export -> Download as CSV -> Observe that only the selected rows are exported, as expected

I can file a separate issue for the selection behavior if you'd like, but I assume it's related to the same implementation inconsistencies that are causing the original issue
